app/models/concerns/foreman_bootdisk/compute_resources/proxmox.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ForemanBootdisk
+  module ComputeResources
+    module Proxmox
+      def capabilities
+        super + [:bootdisk]
+      end
+
+      def iso_upload(iso, vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+        server.ssh_options = { password: fog_credentials[:proxmox_password] }
+        server.ssh_ip_address = server.node_id
+        server.username = client.credentials[:current_user].split('@').first
+        server.scp_upload(iso, '/var/lib/vz/template/iso/')
+        server.reload
+        storage = storages(server.node_id, 'iso')[0]
+        storage.volumes.any? { |v| v.volid.include? File.basename(iso) }
+      end
+
+      def iso_attach(iso, vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+        storage = storages(server.node_id, 'iso')[0]
+        volume = storage.volumes.detect { |v| v.volid.include? File.basename(iso) }
+        disks = server.disks.map { |disk| disk.split(":")[0] }.join(";")
+        config_hash = { ide2: "#{volume.volid},media=cdrom" }
+        server.update(config_hash)
+        server.update({ boot: "order=ide2;#{disks}" })
+        server.reboot
+      end
+
+      def iso_detach(vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+
+        # get volid to delete iso after detaching from vm
+        volid = server.volumes.get("ide2").volid
+        config_hash = { ide2: "none,media=cdrom" }
+        server.update(config_hash)
+
+        # cdrom will be ejected on next power off
+        server.detach('ide2')
```

Comment:
Given the repetition, would it be good to declare `ide2` as a constant?

Reply:
ide2 is a cdrom and for attachment and detachment it has different values. In iso detach function, first it is ejected, thus, set to none and  detach action takes place which will be executed when the vm is powered off. So, it cannot be set as a constant.

Reply:
But it's hardcoded in the code, so it already is. I don't see why you can't use `CDROM_VOLUME = 'ide2'` and then later `server.detach(CDROM_VOLUME)`.